com.unity.ml-agents.extensions/Runtime/Sensors/PoseExtractor.cs

```diff
@@ -165,6 +227,17 @@ public static Pose Multiply(this Pose pose, Pose rhs)
             return rhs.GetTransformedBy(pose);
         }
 
+        /// <summary>
+        /// Transform the vector by the pose.
```

Comment:
I'll try to elaborate in the comment. But basically I think of it as applying a transform to a vector, instead of thinking about spaces.

You can think of a pose as a 4x4 matrix, and multiplying a 4x1 vector

```
    | p.x    rhs.x
  R | p.y    rhs.y
    | p.z  * rhs.z
 ---+----    -----
 000|  1       1
```
where R is the 3x3 rotation matrix that corresponds to the quaternion (see also https://en.wikipedia.org/wiki/Affine_transformation#Augmented_matrix)